crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -516,24 +511,43 @@ mod tests {
 
     #[tokio::test]
     async fn maintenance_can_update_concurrently() {
-        const WAIT_TIME_MS: u64 = 100;
-        const BATCH_SIZE: usize = 100;
-        let mut inner = MockNativePriceEstimating::new();
-        inner
-            .expect_estimate_native_price()
-            .times(BATCH_SIZE)
-            .returning(|_| async { Ok(1.0) }.boxed());
-        // background task updates all outdated prices
-        inner
-            .expect_estimate_native_price()
-            .times(BATCH_SIZE)
-            .returning(move |_| {
-                async move {
-                    tokio::time::sleep(tokio::time::Duration::from_millis(WAIT_TIME_MS)).await;
+        struct OwnMockMockNativePriceEstimating {
```

Comment:
Please, forgive me for this :pray: but there was no way of returning it properly as `mockall` doesn't support `returning` with async inside :disappointed:

Reply:
Nothing happened here :eyes: